image/scripts/app_launch

```diff
@@ -1,19 +1,11 @@
 #!/bin/bash
 
-#Ensure background processes are killed when the script ends, otherwise pebble
-# will never see the process ending and think it's still running
-trap 'trap - SIGTERM && kill 0' SIGINT SIGTERM EXIT
-
-cd "${CONTAINER_APP_ROOT}/app" || exit 1
-
 export UNICORN_BIND_ALL=0.0.0.0
 export UNICORN_SIDEKIQS=1
 
-# Putting the tail's in the background gets the various logs outputting to
-# stdout. The sed allows us to distinguish where a given log line comes form.
-tail -f "${CONTAINER_APP_ROOT}/app/log/production.log" | sed 's/^/discourse: /' &
-tail -f "${CONTAINER_APP_ROOT}/app/log/unicorn-stderr.log" | sed 's/^/unicorn-error: /' 1>&2 &
+su -s /bin/bash -c "${CONTAINER_APP_ROOT}/app/bin/bundle exec prometheus_exporter" &
+su -s /bin/bash -c "${CONTAINER_APP_ROOT}/app/bin/unicorn -c ${CONTAINER_APP_ROOT}/app/config/unicorn.conf.rb" "${CONTAINER_APP_USERNAME}" &
 
-# Add the discourse bin and scripts directories to the PATH.
-export PATH="${CONTAINER_APP_ROOT}/app/bin:${CONTAINER_APP_ROOT}/app/scripts:$PATH"
-su -s /bin/bash -c "${CONTAINER_APP_ROOT}/app/bin/unicorn -c ${CONTAINER_APP_ROOT}/app/config/unicorn.conf.rb" "${CONTAINER_APP_USERNAME}" 2>&1
+# If one of the processes exits, the other one will be killed so that the pod will be restarted by the failing probes
+wait -n
+kill "$(jobs -p)"
```

Comment:
No fan of this coompared to the "trap" method, but that'll do the job
Any reasoning compared to the "trap" ?

Reply:
`trap 'trap - SIGTERM && kill 0' SIGINT SIGTERM EXIT`

This line and its comment you removed from that file

Reply:
The trap command kills all process with the same process group which should work in your case as both processes are part of the same process group and inherits the signal trap

Did you try ?

Reply:
Ok got it